controllers/nodemaintenance_controller.go

```diff
@@ -164,10 +165,12 @@ func (r *NodeMaintenanceReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return r.onReconcileError(instance, fmt.Errorf("Failed to extend lease owned by us : %v errorOnLeaseCount %d", err, instance.Status.ErrorOnLeaseCount))
 	}
 	if err != nil {
+		setLastUpdate(instance)
 		instance.Status.ErrorOnLeaseCount = 0
 		return r.onReconcileError(instance, err)
 	} else {
 		if instance.Status.Phase != nodemaintenancev1beta1.MaintenanceRunning || instance.Status.ErrorOnLeaseCount != 0 {
+			setLastUpdate(instance)
```

Comment:
not needed, will be set again later on

Reply:
in the context of the other comment about the if pendingPods != nil, this one needs to stay then, for setting the timestamp in case drain succeeds on 1st try

Reply:
or better place it at the drain command later on, in case it does not error, because drain can take a while. But there you also need to prevent just updating status... maybe like this:
```
	if err = drain.RunNodeDrain(r.drainer, nodeName); err != nil {
         ...
	} else if instance.Status.Phase != nodemaintenancev1beta1.MaintenanceSucceeded {
          setLastUpdate()
        }
       REMOVE if pendingpods here
```

Reply:
Good points, I think it looks better now.